Rakefile

```diff
@@ -456,6 +448,46 @@ task :authors do
   sh "(git log --use-mailmap --format='%aN <%aE>' ; cat .OLD_AUTHORS) | sort -uf > AUTHORS"
 end
 
+namespace :ci do
+  task :upload_to_sauce do
+    upload_path = ENV['UPLOAD_PATH']
+    username = ENV['SAUCE_USERNAME']
+    apikey = ENV['SAUCE_APIKEY']
+    upload_filename = Pathname.new(upload_path).basename
+    upload_url = "http://saucelabs.com/rest/v1/storage/#{username}/#{upload_filename}"
+    body = nil
+
+    File.open(find_file(upload_path), 'r') do |infile|
+      body = infile.read
+    end
+
+    6.times do
+      uri = URI.parse(upload_url)
+      request = Net::HTTP::Post.new(uri.request_uri)
+      request.basic_auth(username, apikey)
+      request['Content-Type'] = 'application/octet-stream'
+      request.body = body
+      http = net_http.new(uri.host, uri.port)
+      http.read_timeout = 60 * 5 # 5 min
+      begin
+        response = http.request(request)
+      rescue StandardError => e
+        puts "issue uploading file: #{e}"
+        next
+      end
+      metadata = JSON.parse(response.body)
+      local_digest = Digest::MD5.hexdigest(body)
+      if metadata['md5'] == local_digest
+        puts "file successfully uploaded: #{metadata['filename']}"
+      else
+        puts "issues uploading file: #{response.code} - #{response.body}"
+      end
+      break
+    end
+    raise 'failed to upload to saucelabs after numerous retries'
+  end
+end
+
 at_exit do
   if File.exist?(".git") && !Platform.windows?
```

Comment:
Should `Platform.windows?` be replaced with `SeleniumRake::Checks.windows?`?

Reply:
For now no. Because we've got two different helpers.

I want to standardise the checks down the line, but lets leave this as is for now.